DOC: Update from_records docstring to not mention DataFrame as valid input

U-S-jun opened the pull request:
- closes #60307 

Removed mention of DataFrame as an acceptable input type, as it is no longer supported
`
        Convert structured or record ndarray to DataFrame.

        Creates a DataFrame object from a structured ndarray, sequence of
        tuples or dicts.

`

Ensured compatibility with numpydoc format by validating against validate_docstrings.py

U-S-jun committed:
DOC: Update from_records docstring to comply with numpydoc style (GH#60307)

jorisvandenbossche commented on pandas/core/frame.py:
```diff
@@ -2116,7 +2116,7 @@ def from_records(
         Convert structured or record ndarray to DataFrame.
 
         Creates a DataFrame object from a structured ndarray, sequence of
```
```suggestion
        Creates a DataFrame object from a structured ndarray, or sequence of
```

(now the "sequence of tuples or dicts" is the last entry in the enumeration, I think we should add an "or" here for clarity?

jorisvandenbossche reviewed (commented):
Thanks or the PR!

Just a small grammatical comment

U-S-jun committed:
Update pandas/core/frame.py

Added "or" before the last item in the enumeration for improved clarity in the from_records docstring.

Co-authored-by: Joris Van den Bossche <jorisvandenbossche@gmail.com>

mroeschke commented:
Thanks @U-S-jun
